.github/workflows/MainDistributionPipeline.yml

```
@@ -0,0 +1,22 @@
+#
+# This workflow calls the main distribution pipeline from DuckDB to build, test and (optionally) release the extension
+#
+name: Main Extension Distribution Pipeline
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref || '' }}-${{ github.base_ref || '' }}-${{ github.ref != 'refs/heads/main' || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  duckdb-main-pipeline:
+    name: DuckDB Main Pipeline
+    uses: samansmink/duckdb/.github/workflows/_extension_distribution.yml@extension-workflows-2
+    with:
+      duckdb_version: 6855f749756c073bbb5a7abf4109580d45465c45
+      extension_name: quack
+      extension_ref: ${{ github.ref_name }}
+    secrets: inherit
```

```suggestion
    with:
      duckdb_version: 6855f749756c073bbb5a7abf4109580d45465c45
      extension_name: quack
      extension_ref: ${{ github.ref_name }}
    secrets:
      S3_DEPLOY_ID: ${{ env.S3_DEPLOY_ID }}
        required: false
      S3_DEPLOY_KEY: ${{ env.S3_DEPLOY_KEY }}
        required: false
      S3_REGION: ${{ env.S3_REGION }}
        required: false
      S3_BUCKET: ${{ env.S3_BUCKET }}
        required: false
```
I am not completely sure of the syntax, expecially of the `required` part plus what happens if those are absent.
Basically being explicit on what secrets are we using, and also allows arbitrary mappings to be performed here.

---

decided on adding a comment instead here to keep the template clean, but still make people aware the options here